### PR TITLE
[nrf528xx] change directory structure for nrf528xx family

### PR DESCRIPTION
**Pull request by konradderda**
Merged nrf528xx sub-directories to simplify maintenance and porting of new nrf528xx family platforms in the future.

Crucial changes:
- Moved all source files to common 'src' directory
- Merged all *.c files and treated platform specific fragments of code with conditional compilation
- Created simple automake file [examples/platforms/nrf528xx/Makefile.am] that includes conditional-include blocks

To be done:
- Create single README.md file for all nrf528xx platforms

### CODE REVIEW

File: examples/platforms/nrf528xx/src/platform-config.h [hunk 1]
```diff
@@ -26,33 +26,13 @@
...
+ * @file
+ * This file includes the proper platform configuration.
+ */
 
-uint32_t nrf5FlashWrite(uint32_t aAddress, const uint8_t *aData, uint32_t aSize)
-{
-    nrf_nvmc_write_bytes(aAddress, aData, aSize);
-    return aSize;
-}
+#if defined(NRF52811_XXAA)
```
[the remark this answers is not in the document]
Comment (konradderda) [reply examples/platforms/nrf528xx/src/platform-config.h]: Done. We do not need this file now indeed. It was used to fix compilation issues in previous iterations but after clean ups build works fine.

**Comment by konradderda** [comment]
@LuDuda I've created README.md file for nRF528XX platform which contains links to READMEs for nRF52840 and nRF52811. I tried to merge them but It would lead no almost unreadable file. To my mind we should leave it this way now and create a new issue about proper merge of the files.

File: examples/platforms/nrf528xx/Makefile.am [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+include nrf52811/Makefile.am
+endif
+
+if OPENTHREAD_EXAMPLES_NRF52840
+include nrf52840/Makefile.am
+endif
+
+PRETTY_SUBDIRS =     \
+    src              \
+    $(NULL)
```
Comment (jwhui) [inline examples/platforms/nrf528xx/Makefile.am]: nit: missing newline

File: examples/platforms/nrf528xx/nrf52840/Makefile.am [hunk 3]
```diff
@@ -79,35 +80,29 @@ COMMONCPPFLAGS
...
+    src/spi-slave.c                                                                                          \
+    src/system.c                                                                                             \
+    src/temp.c                                                                                               \
+    src/uart.c                                                                                               \
+    src/usb-cdc-uart.c                                                                                       \
     $(NULL)
 
 SINGLEPHY_SOURCES                                                                                          = \
-    flash_nosd.c                                                                                             \
+    src/flash_nosd.c                                                                                             \
```
Comment (jwhui) [inline examples/platforms/nrf528xx/nrf52840/Makefile.am]: nit: align `\` here and below

File: examples/platforms/nrf528xx/src/Makefile.am [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+include $(abs_top_nlbuild_autotools_dir)/automake/pre.am
+
+PRETTY_FILES=                                   \
+    *.c                                         \
+    platform-fem.h                              \
+    platform-nrf5.h                             \
+    $(NULL)
+
+include $(abs_top_nlbuild_autotools_dir)/automake/post.am
```
Comment (jwhui) [inline examples/platforms/nrf528xx/src/Makefile.am]: nit: missing newline

File: examples/platforms/Makefile.am [hunk 5]
```diff
@@ -39,8 +39,8 @@ DIST_SUBDIRS                            = \
     efr32mg21                             \
     gp712                                 \
     kw41z                                 \
-    nrf52811                              \
-    nrf52840                              \
+    nrf528xx                              \
+    nrf528xx/src                          \
```
Comment (jwhui) [inline examples/platforms/Makefile.am]: nit: this should be included in `examples/platforms/nrf528xx/Makefile.am`

File: examples/platforms/nrf528xx/Makefile.am [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+if OPENTHREAD_EXAMPLES_NRF52811
+include nrf52811/Makefile.am
+endif
+
+if OPENTHREAD_EXAMPLES_NRF52840
+include nrf52840/Makefile.am
+endif
+
+PRETTY_SUBDIRS =     \
+    src              \
```
Comment (jwhui) [inline examples/platforms/nrf528xx/Makefile.am]: nit: add `nrf52811` and `nrf52840`?

File: examples/platforms/nrf528xx/nrf52811/Makefile.am [hunk 7]
```diff
@@ -134,4 +127,4 @@ libopenthread_nrf52811_a_LIBADD
 libopenthread_nrf52811_sdk_a_LIBADD                                                                        = \
     $(shell find $(top_builddir)/examples/platforms/utils $(Dash)type f $(Dash)name "*.o")
 
-include $(abs_top_nlbuild_autotools_dir)/automake/post.am
+include $(abs_top_nlbuild_autotools_dir)/automake/post.am
```
Comment (jwhui) [inline examples/platforms/nrf528xx/nrf52811/Makefile.am]: nit:  missing newline

**Comment by konradderda** [comment]
@jwhui I have applied your comments. I've also removed nrf528xx/src/Makefile.am which is not needed now.

**Comment by konradderda** [comment]
@LuDuda Fixed now.

**Pull request merged**